agent/config/builder.go

```diff
@@ -2595,20 +2622,41 @@ func (b *builder) cloudConfigVal(v Config) hcpconfig.CloudConfig {
 		return val
 	}
 
-	val.AuthURL = stringVal(v.Cloud.AuthURL)
-	val.Hostname = stringVal(v.Cloud.Hostname)
-	val.ScadaAddress = stringVal(v.Cloud.ScadaAddress)
+	// Load configuration file variables for anything not set by environment variables
+	if val.AuthURL == "" {
+		if authURL := stringVal(v.Cloud.AuthURL); authURL != "" {
+			val.AuthURL = authURL
+		}
```

Comment:
What do you think about removing the `!= ""` check here (and below) and leaving this as:

```suggestion
		val.AuthURL = stringVal(v.Cloud.AuthURL)
```

It's more readable to me and wouldn't change the overall behavior

Reply:
Oh, yeah, that's true! I'll make this change and also look back to why we were checking this with resource ID (which I'm guilty of just copying the pattern and not thinking about it).

Reply:
To close the loop, it's because we didn't want to override the environment variable with an empty resource ID! But that's taken care of now with checking if the value is already set.